### PR TITLE
MBS-11986 / MBS-12128: New account flag: Spammer

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-11986 / MBS-12128

For now this blocks the user from editing and leaving notes (a combo of editing disabled and notes disabled) and from logging in once logged out.

Future improvements:
* [ ] Automatically log out the editor when marked as spam, so that the login block has a stronger effect. MBS-12129
* [ ] Send user deletion signal to discourse when marked as spam, if desired (but if so, we should figure out a way to restore this if the flag turns out to be added in error). MBS-12130

It was suggested to block profile pages as a 403 but that kinda suggests that there's an issue with the user viewing it - maybe instead detach to a "this user was blocked as spam" template if not account admin? Something like "This user was blocked and their profile is hidden because they were deemed to be spamming. If you see spam in MusicBrainz, please do let us know by reporting the spammer from their user page.".

### CODE REVIEW

**Comment by Freso** [comment]
> Send user deletion signal to discourse when marked as spam, if desired (but if so, we should figure out a way to restore this if the flag turns out to be added in error).

As I mentioned during the meeting, this is currently fairly manageable to do manually. I would probably make a new ticket for this instead of holding this one back on that account (once MBS-111987 gets implemented, this would get a bit more unwieldy :)). Also note that it might be possible to suspend the user instead of deleting them. That would still keep them in the forum db, but would prevent them from doing anything meaningful and would be very easy to restore in case of a mistaken flag.

File: root/admin/edit_user.tt
Comment (Freso) [inline root/admin/edit_user.tt:16]: Maybe put Spammer either at top or bottom? (Also, I had been meaning to make a ticket specifically for this, I just hadn’t gotten around to it yet. Ah well. :joy:)

Comment (reosarevok) [reply root/admin/edit_user.tt:16]: You can still make a ticket and I'll link it from the commit message ;)

**Comment by Freso** [comment]
FWIW, it would be nice to be able to mark accounts as Spam even if it doesn’t do much else than disabling editing and notes. The sooner the flag gets in, the less accounts I’ll have to go back and reflag when it does. :)

File: root/admin/edit_user.tt
Comment (Freso) [reply root/admin/edit_user.tt:16]: https://tickets.metabrainz.org/browse/MBS-12128

File: lib/MusicBrainz/Server/Entity/Editor.pm
Comment (mwiencek) [inline lib/MusicBrainz/Server/Entity/Editor.pm:105]: It's kinda odd how `is_editing_disabled` is now not always equal to the negation of `is_editing_enabled`.  I'd expect that to be true.

Is there any downside to applying the same logic to `is_editing_disabled` too?  Or even `is_adding_notes_disabled` for that matter?

Alternatively, would it make sense to just set `$EDITING_DISABLED_FLAG` and `$ADDING_NOTES_DISABLED_FLAG` automatically whenever the spammer flag is added?

Comment (reosarevok) [reply lib/MusicBrainz/Server/Entity/Editor.pm]: I implemented the latter suggestion as per our IRC chat.

**Review by mwiencek — APPROVED** [review]
Meant to approve this after my last review.

**Review by yvanzo — APPROVED** [review]
Tested by pretending to address spam on my own local server, not difficult to guess who the spammer was. Nothing critical, better merge it now. Please deploy and quickly try it on beta first.

**Pull request merged**
